crates/cheatnet/src/runtime_extensions/forge_runtime_extension/mod.rs

```diff
@@ -54,25 +54,26 @@ pub struct ForgeExtension<'a> {
 }
 
 trait BufferReaderExt {
-    fn read_cheat_target(&mut self) -> CheatTarget;
+    fn read_cheat_target(&mut self) -> BufferReadResult<CheatTarget>;
 }
 
 impl BufferReaderExt for BufferReader<'_> {
-    fn read_cheat_target(&mut self) -> CheatTarget {
-        let cheat_target_variant = self.read_felt().to_u8();
-        match cheat_target_variant {
+    fn read_cheat_target(&mut self) -> BufferReadResult<CheatTarget> {
+        let cheat_target_variant = self.read_felt()?.to_u8();
+
+        Ok(match cheat_target_variant {
             Some(0) => CheatTarget::All,
-            Some(1) => CheatTarget::One(self.read_felt().into_()),
+            Some(1) => CheatTarget::One(self.read_felt()?.into_()),
             Some(2) => {
                 let contract_addresses: Vec<_> = self
-                    .read_vec()
+                    .read_vec()?
                     .iter()
                     .map(|el| ContractAddress::from_(el.clone()))
                     .collect();
                 CheatTarget::Multiple(contract_addresses)
             }
             _ => unreachable!("Invalid CheatTarget variant"),
```

Comment:
I know I approved it but I noticed that this can also panic (e.g. if we add additional enum variants and someone doesn't swtich the backend etc.). Can we fix this?

Reply:
https://github.com/foundry-rs/starknet-foundry/pull/1880